sql: TestShowTenantFingerprintsProtectsTimestamp failed

sql.TestShowTenantFingerprintsProtectsTimestamp [failed](https://tanzanite.cluster.engflow.com/invocations/default/d863d545-fae5-4177-a0f7-5ec8a17834d4?testReportRun=1&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):


```
=== RUN   TestShowTenantFingerprintsProtectsTimestamp
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowTenantFingerprintsProtectsTimestamp3949282097
    test_log_scope.go:81: use -show-logs to present logs inline
    show_fingerprints_test.go:179: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/show_fingerprints_test.go:179
        	Error:      	Received unexpected error:
        	            	TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh due to encountered recently written committed value /Table/47/1/"\xc9"/0 @1711957719.740410065,1): "sql txn" meta={id=b27a321a key=/Table/62/1/0/0 iso=Serializable pri=0.00484297 epo=0 ts=1711957719.772835382,2 min=1711957719.665188781,0 seq=111} lock=true stat=PENDING rts=1711957719.665188781,0 wto=false gul=1711957720.165188781,0
        	Test:       	TestShowTenantFingerprintsProtectsTimestamp
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestShowTenantFingerprintsProtectsTimestamp3949282097
--- FAIL: TestShowTenantFingerprintsProtectsTimestamp (4.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTenantFingerprintsProtectsTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37265

sql.TestShowTenantFingerprintsProtectsTimestamp [failed](https://tanzanite.cluster.engflow.com/invocations/default/8bef18fb-84c5-4cbe-a9b9-cf574e8b81a7?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [2ce93809d799bacc078f8a1d8e24a710dbff4d66](https://github.com/cockroachdb/cockroach/commits/2ce93809d799bacc078f8a1d8e24a710dbff4d66):


```
=== RUN   TestShowTenantFingerprintsProtectsTimestamp
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowTenantFingerprintsProtectsTimestamp4078031098
    test_log_scope.go:81: use -show-logs to present logs inline
    show_fingerprints_test.go:179: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/show_fingerprints_test.go:179
        	Error:      	Received unexpected error:
        	            	TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh due to encountered recently written committed value /Table/47/1/"\xca"/0 @1718092128.820581086,0): "sql txn" meta={id=a1d3bf25 key=/Table/63/1/0/0 iso=Serializable pri=0.01290874 epo=0 ts=1718092128.823513214,2 min=1718092128.747627714,0 seq=129} lock=true stat=PENDING rts=1718092128.747627714,0 wto=false gul=1718092129.247627714,0
        	Test:       	TestShowTenantFingerprintsProtectsTimestamp
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestShowTenantFingerprintsProtectsTimestamp4078031098
--- FAIL: TestShowTenantFingerprintsProtectsTimestamp (4.32s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTenantFingerprintsProtectsTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/show_fingerprints_test.go
+++ b/pkg/sql/show_fingerprints_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigptsreader"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -139,6 +140,11 @@ func TestShowFingerprintsDuringSchemaChange(t *testing.T) {
 func TestShowTenantFingerprintsProtectsTimestamp(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	// Under deadlock there isn't enough time for the CREATE TENANT txn to commit
+	// due to the intervals we lower to speed up this test. There is no difference
+	// otherwise when running this test under deadlock
+	skip.UnderDeadlock(t, 121445, "Deadlock makes txns take too long to commit")
 
 	ctx := context.Background()
 

